tests/api/api_routes/builds/test_build_app.py

```diff
@@ -0,0 +1,168 @@
+from unittest import TestCase
+from unittest.mock import MagicMock, call, patch
+
+from nestor_api.api.flask_app import create_app
+
+
+def _mock_thread(target, args):
+    mock = MagicMock()
+    mock.start.side_effect = lambda: target(*args)
+    return mock
+
+
+# pylint: disable=bad-continuation
+@patch("nestor_api.api.api_routes.builds.build_app.Thread", side_effect=_mock_thread)
+@patch("nestor_api.api.api_routes.builds.build_app.Logger", autospec=True)
+@patch("nestor_api.api.api_routes.builds.build_app.app", autospec=True)
+@patch("nestor_api.api.api_routes.builds.build_app.config", autospec=True)
+@patch("nestor_api.api.api_routes.builds.build_app.docker", autospec=True)
+@patch("nestor_api.api.api_routes.builds.build_app.git", autospec=True)
+@patch("nestor_api.api.api_routes.builds.build_app.io", autospec=True)
+class TestApiBuildApp(TestCase):
+    def setUp(self):
+        self.app_client = create_app().test_client()
+
+    @patch("nestor_api.api.api_routes.builds.build_app.Configuration", autospec=True)
+    def test_build_app(
+        self,
+        configuration_mock,
+        io_mock,
+        git_mock,
+        docker_mock,
+        config_mock,
+        app_mock,
+        logger_mock,
+        _thread_mock,
+    ):
+        # Mock
+        configuration_mock.get_config_default_branch.return_value = "default"
+
+        app_config = {
+            "git": {"origin": "git@github.com:my-org/my-app.git"},
+            "workflow": ["master", "production"],
+        }
+        config_mock.create_temporary_config_copy.return_value = "/tmp/config"
+        config_mock.get_app_config.return_value = app_config
+
+        git_mock.create_working_repository.return_value = "/tmp/working/repo"
+
+        app_mock.get_version.return_value = "1.0.0"
+        git_mock.tag.return_value = "1.0.0-sha-a1b2c3d4"
+
+        docker_mock.build.return_value = "my-app@1.0.0-sha-a1b2c3d4"
```

Comment:
Could we have this in the setup and only override needed value for error test cases ?
I would be easier to read IMO.